treemacs.el

```diff
@@ -605,6 +605,40 @@ Delete all elements whose car is ‘eq’ to KEY from ALIST."
   (beginning-of-line)
   (forward-char))
 
+(defun treemacs--path-in-dir? (path dir)
+  "Is PATH in directory DIR?"
+  (s-starts-with? (concat dir "/") path))
```

Comment:
Is adding the slash needed? I can't think of a case where it would make a difference. 
Being this short this function is also a good candidate for being a defsubst instead of a defun.

Everything else looks good. Just squash it up and we're good to go. Since we're adding a single feature IMO the changes should into one commit, or at most two as was the case originally.